### PR TITLE
Add HTML styleguide

### PR DESCRIPTION
**Pull request by MaxWofford**

### CODE REVIEW

**Comment by paked** [comment]
This needs debate

File: styleguides/html.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+_This styleguide has been adapted from
+https://google.github.io/styleguide/htmlcssguide.xml._
+
+-------------------------------------------------------------------------------
+
+# HTML Styleguide
+
+This is a collection of HTML guidelines that Hack Club workshops should follow
```
Comment (paked) [inline styleguides/html.md]: `should` sounds optional

**Comment by paked** [comment]
- We should include a section on "where to put stylesheets"
- We should include a section on "where to put JavaScript includes"
- Should we use minimised resources?
- Do we include resources in the git repository, or defer to a CDN when available

**Comment by paked** [comment]
@MaxWofford Those are the next things we need to look at <3

**Comment by paked** [comment]
@zachlatta would really appreciate your opinion on some of the topics raised above

File: styleguides/html.md (hunk 1 above)
Comment (MaxWofford) [inline styleguides/html.md]: @paked I think having `should` is important because there may be situations that these guides can't be followed.

**Comment by zachlatta** [comment]
Definitely! Happy to chime in :-).

> - We should include a section on "where to put stylesheets"

Stylesheets in hacker source code? Or in the workshop source code? Or for both?

> - We should include a section on "where to put JavaScript includes"

Like `script` tags?

> - Should we use minimised resources?

Without knowing more, my first inclination is that the benefit of standardizing
things like this is marginal. I'm not sure if there's a big enough win to add
_one more requirement_ to the process of making workshops.

> - Do we include resources in the git repository, or defer to a CDN when available

Since this impacts the existence of files in the repository, I'd say this is at
least including recommendations for. My personal opinion is that we should defer
to CDNs when possible and include the source in the repository when we create
them, unless they're an entirely separate project (for things like
`twilio-basic`).

File: styleguides/html.md (hunk 1 above)
Comment (zachlatta) [inline styleguides/html.md]: What if we changed it to the following?

> This is a collection of HTML guidelines for Hack Club workshops

I think that leaves enough ambiguity to have occasional deviations, but it's
also strongly worded enough to make it clear that these are the default
guidelines for workshops.

**Comment by paked** [comment]
@zachlatta 

Okay, rephrasing the first two points: 

> Where do we as authors put our JavaScript and CSS links (respectively `<script>` and `<link>` tags)

Possible options could be:

- Wherever it doesn't matter
- At the bottom of the `<head>` tag
- At the bottom of the `<body>` tag

I personally think this would be valuable to standardise so that hackers know *exactly* where to look when they want to find these "includes".

File: styleguides/html.md (hunk 1 above)
Comment (paked) [inline styleguides/html.md]: :+1:

**Comment by paked** [comment]
@zachlatta ping

**Comment by zachlatta** [comment]
Ah, sorry for the delay!

Some of this will definitely be contextual though, right? For some
JavaScript, we may want it to load before the page renders, so we'd put it
in <head>. For most JavaScript, we probably don't mind if it doesn't load
until after the page renders, so we'd put it in <body>. I'd bias for
"Wherever it doesn't matter" for the time being and not worry about
clarifying that in the styleguide (at least not yet).

**Comment by paked** [comment]
Thanks for discussing @zachlatta @MaxWofford :D.

@jonleung Lets try and implement this with the portfolio rewrite.

**Pull request merged**